libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/FileAccessTree.java

```diff
@@ -179,9 +184,10 @@ public static FileAccessTree of(
         String moduleName,
         FilesEntitlement filesEntitlement,
         PathLookup pathLookup,
+        @Nullable Path componentDir,
```

Comment:
Instead of this being a "dir" how about just "componentPath"? If it's a plugin, it's the plugin dir. If it's a jar from server, it's the jar path. This way modules that server depends on (say 3rd party libs that do this self reflection) can still access their own jar.

Reply:
If it's a jar from server, does it have to be the jar of the calling class, or can it be libDir?
(should we do `Paths.get(requestingClass.getProtectionDomain().getCodeSource().getLocation().toURI())` for every server check?)

Reply:
It could be, but ultimately why be so broad? I'm not sure we even need the entire plugin directory? It could just be each module has access to its own jar. The argument for the plugin directory is if the plugin contains other resource files, but those could be within the plugin's jar itself.